code/my_set/viewmodel/viewmodel.py

```diff
@@ -0,0 +1,65 @@
+from my_set.model.my_set import Set
+
+
+class SetViewModel:
+    def __init__(self):
+        self.set_A = Set()
+        self.set_B = Set()
+        self.result = None
+        self.error_msg = None
+
+    def add(self, data, name_set):
+        try:
+            if name_set == 'A':
+                processed_data = eval(data)
+                self.set_A.add(processed_data)
+            else:
+                processed_data = eval(data)
+                self.set_B.add(processed_data)
+        except Exception as exp:
+            self.error_msg = "Error: {}". format(exp)
```

Comment:
Now, when value of incorrect type entered (e. g. `'a'`) the message shown is `Error: name 'a' is not defined`. Probably it is better not to show the user internal error messages, but notify that the entered value is of incorrect type.

Reply:
Done.